File: NeetCode-150/Arrays and Hashing/Two Sum/solution.cpp
// Hari

vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> umap;
        vector<int> res;
        
        for(int i = 0; i<nums.size(); i++) {
            umap[nums[i]] = i;
        }
        
        for(int i = 0; i<nums.size(); i++) {
            if (umap.find(target - nums[i]) != umap.end() &&
                umap[target - nums[i]] != i) {
                res.push_back(i);
                res.push_back(umap[target - nums[i]]);
                break;
            }
        }
        
        return res;
    }

File: NeetCode-150/Arrays and Hashing/Two Sum/solution.java
// Sid 
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] res = new int[2];
        Map<Integer, Integer> mp = new HashMap<Integer, Integer>(); 

        for(int i = 0; i < nums.length; i++){
            mp.put(nums[i], i); 
        }

        for(int i = 0; i < nums.length; i++){
            int tar = target - nums[i]; 
            if(mp.get(tar) != null && mp.get(tar) != i){
                res[0] = i; 
                res[1] = mp.get(tar); 
                break;
            }
        }

        return res;
    }
}
